File: web/vue/src/ai/api/conversation.ts
/**
 * 会话 API 函数
 *
 * 与后端 AI 服务的会话接口通信
 */
import { del, get } from "@/framework/api/client";
import type { ApiResponse } from "@/framework/api/types";

/**
 * 会话列表项（与后端 Schema 一致）
 */
export interface ConversationListItem {
  id: string;
  name: string;
  created_at: string;
  message_count: number;
}

/**
 * 会话列表响应
 */
export interface ConversationListResponse {
  conversations: ConversationListItem[];
}

/**
 * 获取会话列表
 */
export const getConversations = (): Promise<ConversationListResponse> =>
  get<ApiResponse<ConversationListResponse>>("/ai/console/v1/conversations").then((res) => res.data!);

/**
 * 删除会话
 */
export const deleteConversation = (id: string): Promise<{ success: boolean }> =>
  del<ApiResponse<{ success: boolean }>>(`/ai/console/v1/conversations/${id}`).then((res) => res.data!);

File: server/python/src/ai/schemas/conversation.py
"""会话管理 Schema

定义会话列表和删除接口的请求/响应 Schema。
"""

from datetime import datetime

from framework.schemas import BaseModel
from pydantic import Field


class ConversationListItem(BaseModel):
    """会话列表项 Schema"""

    id: str = Field(description="会话 ID")
    name: str = Field(description="会话名称")
    created_at: datetime = Field(description="创建时间")
    message_count: int = Field(default=0, description="消息数量")


class ConversationListResponse(BaseModel):
    """会话列表响应 Schema"""

    conversations: list[ConversationListItem] = Field(default_factory=list, description="会话列表")


class ConversationDeleteResponse(BaseModel):
    """会话删除响应 Schema"""

    success: bool = Field(description="是否成功")
